README.md

```diff
@@ -587,6 +601,231 @@ that you get for free from OverReact, you're ready to start building your own cu
+  unnecessary null-checking in your business logic. 
+
+  Also, if you fail to provide an initial value for all the fields in 
+  `state`, your component will fail to build at run-time.
+
+  _Good:_
+    ```dart
+    @Props()
+    DropdownButtonProps extends UiProps {
+      /// Whether the [DropdownButton] appears disabled.
+      /// 
+      /// Default: `false`
+      bool isDisabled;
+
+      /// Whether the [DropdownButton]'s child [DropdownMenu] is open
+      /// when the component is first mounted.
+      /// 
+      /// Determines the initial value of [DropdownButtonState.isOpen].
+      /// 
+      /// Default: `false`
+      bool initiallyOpen;
+    }
+
+    @State()
+    DropdownButtonState extends UiState {
+      /// Whether the [DropdownButton]'s child [DropdownMenu] is open.
+      /// 
+      /// Initial: [DropdownButtonProps.initiallyOpen]
+      bool isOpen;
+    }
+
+    @Component()
+    DropdownButtonComponent 
+        extends UiStatefulComponent<DropdownButtonProps, DropdownButtonState> {
+      @override
+      Map getDefaultProps() => (newProps()
+        ..isDisabled = false
+        ..initiallyOpen = false
+      );
+
+      @override
+      Map getInitialState() => (newState()
+        ..isOpen = props.initiallyOpen
+      );
+    }
+    ```
+
+  _Bad:_
+    ```dart
+    @Props()
+    DropdownButtonProps extends UiProps {
+      bool isDisabled;
+      bool initiallyOpen;
+    }
+
+    @State()
+    DropdownButtonState extends UiState {
+      bool isOpen;
+    }
+
+    @Component()
+    DropdownButtonComponent 
+        extends UiStatefulComponent<DropdownButtonProps, DropdownButtonState> {
+      // Confusing stuff is gonna happen in here with bool props that could
+      // be null, and un-initialized state keys.
+    }
+    ```
+
+&nbsp;
+
+* __AVOID__ adding `props` or `state` fields that don't have 
+an informative comment.
+
+  _Good:_
+    ```dart
+    @Props()
+    DropdownButtonProps extends UiProps {
+      /// Whether the [DropdownButton] appears disabled.
+      /// 
+      /// Default: `false`
+      bool isDisabled;
+
+      /// Whether the [DropdownButton]'s child [DropdownMenu] is open
+      /// when the component is first mounted.
+      /// 
+      /// Determines the initial value of [DropdownButtonState.isOpen].
+      /// 
+      /// Default: `false`
+      bool initiallyOpen;
+    }
+
+    @State()
+    DropdownButtonState extends UiState {
+      /// Whether the [DropdownButton]'s child [DropdownMenu] is open.
+      /// 
+      /// Initial: [DropdownButtonProps.initiallyOpen]
+      bool isOpen;
+    }
+    ```
+
+  _Bad:_
+    ```dart
+    @Props()
+    DropdownButtonProps extends UiProps {
+      bool isDisabled;
+      bool initiallyOpen;
+    }
+
+    @State()
+    DropdownButtonState extends UiState {
+      bool isOpen;
+    }
+    ```
+
+&nbsp;
+
+* __AVOID__ specifying more than one cascading prop setter on the same line.
```

Comment:
This should be grouped with the first couple guidelines.

Also, would it make sense to have separate sections for guidelines around consuming components and guidelines around building components?